web/tests/integration/app.test.ts

```diff
@@ -0,0 +1,302 @@
+import { gql } from "@apollo/client";
+
+import {
+  integrationDBExecuteQuery,
+  integrationDBSetup,
+  integrationDBTearDown,
+} from "./setup";
+
+import { getAPIUserClient } from "./test-utils";
+import { generateAPIKeyJWT } from "@/backend/jwts";
+import getConfig from "next/config";
+const { publicRuntimeConfig } = getConfig();
+// TODO: Consider moving this to a generalized jest environment
+beforeEach(integrationDBSetup);
+beforeEach(integrationDBTearDown);
+
+describe("user role", () => {
+  test("cannot select another team's apps", async () => {
+    const { rows: teams } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."team"`
+    )) as { rows: Array<{ id: string }> };
+
+    for (const team of teams) {
+      const { rows: teamApps } = (await integrationDBExecuteQuery(
+        `SELECT id FROM "public"."app" WHERE "team_id" = '${team.id}';`
+      )) as { rows: Array<{ id: string }> };
+
+      const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+        `SELECT id, user_id FROM "public"."membership" WHERE "team_id" = '${team.id}' limit 1;`
+      )) as { rows: Array<{ id: string; user_id: string }> };
+
+      const client = await getAPIUserClient({
+        team_id: team.id,
+        user_id: teamMemberships[0].user_id,
+      });
+
+      const query = gql(`query ListApps {
+        app(order_by: {created_at: asc}) {
+          id
+          team_id
+        }
+      }`);
+
+      const response = await client.query({ query });
+      expect(response.data.app.length).toEqual(teamApps.length);
+
+      response.data.app.forEach((app: { id: string; team_id: string }) => {
+        expect(app.team_id).toEqual(team.id);
+        expect(teamApps.map((app) => app.id)).toContain(app.id);
+      });
+    }
+    // Use an incorrect team
+    const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+      `SELECT id, user_id FROM "public"."membership" WHERE "team_id" = '${teams[1].id}' limit 1;`
+    )) as { rows: Array<{ id: string; user_id: string }> };
+    const client = await getAPIUserClient({
+      team_id: teams[0].id,
```

Comment:
worth creating a single test case on `app.test.ts` for what happens if you pass a completely invalid team ID